controllers/networking-calico/pkg/apis/calico/v1alpha1/types_network.go

```diff
@@ -52,9 +61,15 @@ type NetworkConfig struct {
 	// https://docs.projectcalico.org/v2.2/reference/node/configuration#ip-autodetection-methods
 	// +optional
 	IPAutoDetectionMethod *string `json:"ipAutodetectionMethod,omitempty"`
-	// IPIP is the IPIP Mode for the IPv4 Pool (e.g. Always, Never, CrossSubnet)
+	// IPv4Pool configures the type of ip pool for the tunnel interface
+	// see: https://docs.projectcalico.org/v3.8/reference/node/configuration#environment-variables
+	// +optional
+	IPv4Pool *IPv4Pool `json:"ipv4pool,omitempty"`
+	// IPv4PoolMode is the mode for the IPv4 Pool (e.g. Always, Never, CrossSubnet)
+	// IPIP pools accept all IPIP values
+	// VXLAN pools accept only Always and Never (unchecked)
 	// +optional
-	IPIP *IPIP `json:"ipip,omitempty"`
+	IPv4PoolMode *IPv4PoolMode `json:"ipv4poolmode,omitempty"`
```

Comment:
```suggestion
	IPv4PoolMode *IPv4PoolMode `json:"ipv4PoolMode,omitempty"`
```

Reply:
Though, instead of having both `ipv4Pool` and `ipv4PoolMode`, why not only having `ipv4` that has a `pool` and a `mode` field?

```yaml
...
ipv4:
  pool: {IPIP,VXLAN}
  mode: {Always,Never,CrossSubnet,Off}
```

Reply:
We did what you suggested.